test/e2e/k8s_util.go

```diff
@@ -103,6 +103,7 @@ type TestNamespaceMeta struct {
 // GetPodByLabel returns a Pod with the matching Namespace and "pod" label if it's found.
 // If the pod is not found, GetPodByLabel returns "ErrPodNotFound".
 func (k *KubernetesUtils) GetPodByLabel(ns string, name string) (*v1.Pod, error) {
+	//TODO: Query why this is hard coded as "pod"
```

Comment:
this function (and actually most functions in this file, for historical reasons) was meant to be used specifically for matrix-based NetworkPolicy tests. You don't need to use it in your case, and you can remove your comment.